src/coreclr/gc/gc.cpp

```diff
@@ -45747,8 +45747,8 @@ HRESULT GCHeap::Initialize()
         }
         else
         {
-            // If no hard_limit is configured the reservation size is max of 256gb or 2x physical limit
-            gc_heap::regions_range = max(((size_t)256 * 1024 * 1024 * 1024), (size_t)(2 * gc_heap::total_physical_mem));
+            // If no hard_limit is configured the reservation size is max of half of GetVirtualMemoryLimit() or 2x physical limit
+            gc_heap::regions_range = max(GCToOSInterface::GetVirtualMemoryLimit()/2, (size_t)(2 * gc_heap::total_physical_mem));
```

Comment:
```suggestion
            gc_heap::regions_range = max(min(GCToOSInterface::GetVirtualMemoryLimit()/2, (size_t)256 * 1024 * 1024 * 1024), (size_t)(2 * gc_heap::total_physical_mem));
```

Reply:
Thanks.

Reply:
My code is actually wrong. In case the machine happen to have physical memory more than the address space size. This can be a problem.

Can we change it to `min(GCToOSInterface::GetVirtualMemoryLimit()/2, max((size_t)256 * 1024 * 1024 * 1024), (size_t)(2 * gc_heap::total_physical_mem))` instead?

Sorry for the inconvenience.